# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/Pivot.h
# Commit: 892b785f0fc5
# Full Hash: 892b785f0fc5126b67898f2534789160b89e8744
# Author: Morgan Rae Reschenberg <mreschenberg@mozilla.com>
# Date: 2022-10-28 04:43:55
# Regressor Bug: 1787284
# File Overlap Count: 1
# Description:
#   Bug 1787284: [Part 2] Cache name and create radio name pivot rule r=eeejay
#   
#   Depends on D158892
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D159118
# ==============================================================================

diff -r d3d8e4d17161 -r 892b785f0fc5 accessible/base/Pivot.h
--- a/accessible/base/Pivot.h	Thu Oct 27 20:32:17 2022 +0000
+++ b/accessible/base/Pivot.h	Thu Oct 27 20:32:18 2022 +0000
@@ -122,6 +122,20 @@
   virtual uint16_t Match(Accessible* aAcc) override;
 };
 
+/**
+ * This rule matches remote radio button accessibles with the given name
+ * attribute. It assumes the cache is enabled.
+ */
+class PivotRadioNameRule : public PivotRule {
+ public:
+  explicit PivotRadioNameRule(const nsString& aName);
+
+  virtual uint16_t Match(Accessible* aAcc) override;
+
+ protected:
+  const nsString& mName;
+};
+
 }  // namespace a11y
 }  // namespace mozilla
 